.github/workflows/build.yml

```diff
@@ -14,37 +36,53 @@ env:
   CCACHE_KEY_SUFFIX: r1
 
 jobs:
-  find-all-packages:
+  generate-matrices:
     runs-on: ubuntu-22.04
     outputs:
-      packages: ${{ steps.list-packages.outputs.packages }}
+      packages-with-dependents: ${{ steps.list-packages-with-dependents.outputs.packages }}
     steps:
       - name: checkout
         uses: actions/checkout@v4
+        with:
+          # Fetch whole git history.
+          # Required to parse modified packages
+          fetch-depth: 0
 
       - name: setup python 3.12
         uses: actions/setup-python@v5
         with:
           python-version: 3.12
       - name: install python dependencies
-        run: pip install --upgrade pip pyyaml
+        run: pip install --upgrade pip conan
+
+      - name: conan config
+        run: conan config install .github/config/ubuntu-22.04/conan
 
-      - name: List all packages
-        run: python scripts/list_package_versions.py
+      - name: Conan export all packages
+        run: python scripts/conan_export_all_packages.py
+
+      # Generate dependency graphs for all disjoint packages
+      - run: conan graph info recipes/odrcore/all/conanfile.py --version=4.0.0 --profile:host=android-21-armv8 --format=json | tee odrcore-dependency-tree.json
```

Comment:
why `android-21-armv8` specifically?

Reply:
Yes, turns out this is not exactly self-explanatory. Added a comment to the workflow:

Dependency graphs for android-21 and android-23 profiles are not identical.
android-21 has an extra openlibm dependency. Android ABI does not matter for dependency graph, as long as it's still Android.
Generating graphs for both 21 and 23 would be more self-explanatory, but conan graph generation takes a few seconds and I would rather have a comment than waste seconds on each run